server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -156,8 +159,10 @@ private void removeCandidatesInUse(GarbageCollectionEnvironment gce,
 
         dir = makeRelative(dir, 2);
 
-        if (candidateMap.remove(dir) != null) {
+        GcCandidate gcTemp = candidateMap.remove(dir);
+        if (gcTemp != null) {
           log.debug("Candidate was still in use: {}", dir);
+          inUseCandidates.add(gcTemp);
```

Comment:
This is a set, so its possible the add is unsuccessful.  Would we ever expect the add to fail?  wonder if the return value of the add call should be checked or if a list should/could be used instead of a set.

Reply:
Switched to list.